src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2285,112 +2285,190 @@ LogicalResult ONNXPadOp::inferShapes(
 
-LogicalResult ONNXSqueezeV11Op::inferShapes(
+void updateSqueezeOpNegativeAxis(ONNXSqueezeV11Op *op, ArrayRef<int64_t> axes) {
+  updateNegativeAxisV11(op, axes);
+}
+
+template <typename Op, typename Adaptor, typename ShapeHelper>
+LogicalResult ONNXSqueezeOpInferShapesCommon(Op *op,
+    llvm::Optional<ArrayAttr> axisAttrs,
     std::function<void(mlir::Region &)> doShapeInference) {
-  if (!data().getType().isa<RankedTensorType>())
+  if (!op->data().getType().template isa<RankedTensorType>())
     return success();
 
-  auto operandTy = data().getType().cast<RankedTensorType>();
-  auto elementType = data().getType().cast<ShapedType>().getElementType();
+  auto operandTy = op->data().getType().template cast<RankedTensorType>();
+  auto elementType =
+      op->data().getType().template cast<ShapedType>().getElementType();
   int64_t inRank = operandTy.getRank();
 
-  ArrayAttr axisAttrs = axesAttr();
   if (!axisAttrs)
-    return emitError("Axes attribute is required");
+    return op->emitError("Axes attribute is required");
 
   SmallVector<int64_t, 4> axes;
   bool hasNegativeAxis = false;
   for (auto axisAttr : axisAttrs.getValue()) {
     int64_t axis = axisAttr.cast<IntegerAttr>().getInt();
     if (axis < -inRank || axis >= inRank)
-      return emitError("Invalid axis value");
+      return op->emitError("Invalid axis value");
     if (axis < 0) {
       axis = inRank + axis;
       hasNegativeAxis = true;
     }
     if (std::find(axes.begin(), axes.end(), axis) != axes.end())
-      return emitError("Duplicated axes");
+      return op->emitError("Duplicated axes");
     axes.emplace_back(axis);
   }
+
   if (hasNegativeAxis) {
-    // Update axes attribute so that it contains only positive values.
-    auto builder = mlir::Builder(getContext());
-    ArrayRef<int64_t> defaultRefs(axes);
-    axesAttr(builder.getI64ArrayAttr(defaultRefs));
+    updateSqueezeOpNegativeAxis(op, axes);
   }
 
-  ONNXSqueezeV11OpAdaptor operandAdaptor(*this);
-  ONNXSqueezeV11OpShapeHelper shapeHelper(this);
+  Adaptor operandAdaptor(*op);
+  ShapeHelper shapeHelper(op);
   if (failed(shapeHelper.Compute(operandAdaptor)))
-    return emitError("Failed to scan Squeeze parameters successfully");
+    return op->emitError("Failed to scan Squeeze parameters successfully");
   SmallVector<int64_t, 4> outputDims;
   IndexExpr::getShape(shapeHelper.dimsForOutput(0), outputDims);
-  getResult().setType(RankedTensorType::get(outputDims, elementType));
+  op->getResult().setType(RankedTensorType::get(outputDims, elementType));
 
   return success();
 }
 
+LogicalResult ONNXSqueezeOp::inferShapes(
+    std::function<void(mlir::Region &)> doShapeInference) {
+  auto builder = mlir::Builder(getContext());
+  llvm::Optional<ArrayAttr> optionalAttr;
+  if (auto axesConstOp = getONNXConstantOp(axes())) {
+    auto axesAttr = createArrayAttrFromConstantOp(builder, axesConstOp);
+    optionalAttr.emplace(axesAttr);
+  } else if (!axes().getType().isa<NoneType>()) {
```

Comment:
For V13, the empty axes means to reduce every dim with size 1. onnx model generated with newer protobuf, empty axes is represented with Constant of 0 size. You can use `isFromNone(axes())` to check the condition.

Reply:
I pushed a commit to handle empty axes for Squeeze. Let me know if it meets your expectations. Thanks!